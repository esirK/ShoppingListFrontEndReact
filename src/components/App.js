import React, { Component } from 'react';
import RegisterForm from '../containers/register_form'

export default class App extends Component {
  render() {
    return (
      <div className="container">
        <RegisterForm />
      </div>
    );
  }
}