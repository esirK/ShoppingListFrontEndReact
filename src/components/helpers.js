export function validate(values){
	const errors = {};
	if (!values.username){
		errors.username = 'Username Required';
	}
	if (!values.email){
		errors.email = 'Email Required';
	}
	if (!values.password){
		errors.password = 'Password Required';
	}
	if (values.password && values.password.length < 6){
		errors.password = 'Passwords must be at least 6 Characters Long';
	}
	if (!values.confirm_password){
		errors.confirm_password = 'Enter confirmation Password';
	}
	if(values.password !== values.confirm_password){
		console.log('Look at me ',values.confirm_password);
		errors.confirm_password = 'Passwords do not match';
	}
	return errors;
}
