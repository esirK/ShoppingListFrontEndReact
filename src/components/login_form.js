import React, {Component} from 'react';


export default class LoginForm extends Component{
    render(){
        return(
            <div>
                <p>I am Mr Meesix Look at me</p>
            </div>
        );
    }
}