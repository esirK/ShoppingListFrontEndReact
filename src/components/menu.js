import React, {Component} from 'react';
import {connect} from 'react-redux';

import MenuItem from 'material-ui/MenuItem';
import TextField from 'material-ui/TextField';
import {FlatButton} from 'material-ui';

import Dialog from 'material-ui/Dialog';
import Snackbar from 'material-ui/Snackbar';

import IconMenu from 'material-ui/IconMenu';
import IconButton from 'material-ui/IconButton';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

import {Logout, update_profile, hideSnackBar} from '../actions';

export class Menu extends Component{
	constructor(props){
		super(props);
		this.state ={
			conf_logout: false,
			open_profile: false,
			username: '',
			password: '',
			old_password: '',
			openSb: false
		};
		this.logout = this.logout.bind(this);
		this.profile = this.profile.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleLogout = this.handleLogout.bind(this);
		this.handlePasswordChange = this.handlePasswordChange.bind(this);
		this.handleUsernameChange = this.handleUsernameChange.bind(this);
		this.handleProfileUpdate = this.handleProfileUpdate.bind(this);
	}
	logout(){
		this.setState({conf_logout: true});
	}
	profile(){
		this.setState({open_profile: true});
		this.props.hideSnackBar();
	}
	handleLogout(){
		this.props.Logout();
		this.setState({conf_logout: false});
		//reload page on Logout
		window.location.reload();
	}
	handlePasswordChange(e){
		this.setState({password: e.target.value});
	}
	handleUsernameChange(e){
		this.setState({username: e.target.value});
	}
	handleProfileUpdate(){
		this.props.update_profile({username:this.state.username, password:this.state.password});
		this.setState({open_profile: false});
		this.setState({'username':''});
		this.setState({'password':''});
		this.setState({'old_password':''});
	}
	handleClose(){
		this.setState({conf_logout: false});
		this.setState({open_profile: false});
		this.setState({'username':''});
		this.setState({'password':''});
		this.setState({'old_password':''});
		this.props.hideSnackBar();
	}
	render(){
		let actions = [
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.handleClose}
			/>,
			<FlatButton
				label="OK"
				primary={true}
				onClick={this.handleLogout}
			/>,
		];
		let profile_actions = [
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.handleClose}
			/>,
			<FlatButton
				label="Update"
				primary={true}
				onClick={this.handleProfileUpdate}
			/>,
		];
		const profile_children = [
			<TextField
			  hintText="New Username"
			  value={this.state.username}
			  fullWidth={true}
			  key='name'
			  onChange={this.handleUsernameChange}
			/>,
			<TextField
			  hintText="Old Password"
			  multiLine={true}
			  fullWidth={true}
			  value={this.state.password}
			  key='old_p'
			  onChange={this.handlePasswordChange}
			  type={'password'}
			/>,
			<TextField
			  hintText="New Password"
			  multiLine={true}
			  fullWidth={true}
			  key='new_p'
			  onChange={this.handlePasswordChange}
			  type="password"
			>
			</TextField>
		];
		return(
			<div>
				<IconMenu
					iconButtonElement={
						<IconButton><MoreVertIcon /></IconButton>
					}
					targetOrigin={{horizontal: 'left', vertical: 'top'}}
					anchorOrigin={{horizontal: 'right', vertical: 'top'}}
				>
					<MenuItem primaryText="Profile" onClick={this.profile} />
					<MenuItem primaryText="Sign out" onClick={this.logout}/>
				</IconMenu>
				<Dialog
					actions={actions}
					modal={false}
					open={this.state.conf_logout}
					onRequestClose={this.handleClose}
				>
         			Confirm Logout?
				</Dialog>
				<Dialog open={this.state.open_profile}
					title="Update Personal details"
					actions={profile_actions}
					children={profile_children}
					onRequestClose={this.handleClose}
				>
				</Dialog>
				<Snackbar
					open={this.props.openSb}
					message={this.props.message}
					autoHideDuration={3000}
					onRequestClose={this.handleRequestClose}
				/>
			</div>
		);
	}
}
function mapStateToProps(state) {
	return{
		message: state.authentication.message,
		openSb: state.authentication.openSb,
	};
}
export default connect(mapStateToProps, {Logout, update_profile, hideSnackBar})(Menu);