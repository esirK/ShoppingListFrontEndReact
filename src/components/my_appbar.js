import React, { Component } from 'react';

import {connect} from 'react-redux';
import AppBar from 'material-ui/AppBar';

import Menu from './menu';

export class MyAppBar extends Component{
	render(){
		return(
			<AppBar title="Shopping List App" showMenuIconButton={false}
				iconElementRight={this.props.isAuthenticated?<Menu />:null}
			/>
		);
	};
}
function mapStateToProps(state, ownProps) {
	return{
		isAuthenticated: state.authentication.isAuthenticated,
	};
}
export default connect(mapStateToProps, null)( MyAppBar);