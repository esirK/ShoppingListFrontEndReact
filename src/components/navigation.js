import React, { Component } from 'react';
import { Link } from 'react-router-dom';


export default class Navigation extends Component{
	render(){
		return(
			<div className="container">
				<div className="navbar-header">
					<a className="navbar-brand" href="/" title="Awesome Shopping list app">

					</a>
				</div>
				<div className="navbar-collapse ">
					<ul>
						<li>
							<Link to="/">
                                Home
							</Link>
						</li>
					</ul>
				</div>
			</div>
		);
	}
}