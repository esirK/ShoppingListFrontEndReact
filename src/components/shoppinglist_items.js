import React, { Component } from 'react';
import { connect } from 'react-redux';
import { viewShoppingList } from '../actions/index';
import { Card, CardHeader, CardText, CardActions, FlatButton } from 'material-ui';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';

import Dialog from 'material-ui/Dialog';
import Snackbar from 'material-ui/Snackbar';


import NewShoopingListItem from './new_shoppinglist_item';
import UpdateShoppingListItem from './update_shoppinglist_item';
import {activateAddItem, activateUpdateItem, hideSnackBar, deleteShoppinglistItem, resetErrors} from '../actions';

class ShoppingListItems extends Component{
	constructor(props){
		super(props);
		this.state={
			conf_delete: false,
			id: -1
		};
		this.handleFabClick = this.handleFabClick.bind(this);
		this.confirmDelete = this.confirmDelete.bind(this);
		this.handleDelete = this.handleDelete.bind(this);
		this.handleClose = this.handleClose.bind(this);
	}
	componentDidMount(){
		//send request to shoppinglist api for shoppinglist 
		// whose id apear in url
		this.props.viewShoppingList(this.props.match.params.id);
	}
	handleFabClick(){
		//Open a Dialog to add a new shoppinglist Item
		//Sets the openAddItem props to true
		this.props.activateAddItem();
	}
	confirmDelete(id){
		this.setState({
			conf_delete: true,
			id
		});
	}
	handleDelete(id){
		console.log('Delete Item ', this.state.id);
		this.props.deleteShoppinglistItem(this.state.id);
		this.setState({
			conf_delete: false,
			id:-1
		});
		//Reload 
		this.props.viewShoppingList(this.props.match.params.id);
	}
	handleClose(){
		this.setState({
			conf_delete: false,
		});
	}
	render(){
		if(this.props.openAddItem){
			return(
				<NewShoopingListItem {...this.props}/>
			);
		}
		//Initialize empty cards to hold the shoppinglist items
		let cards = [];
		let fab = (
			<div id="cards">
				{cards}
				<div id="fab">
					<FloatingActionButton secondary={true} onClick={this.handleFabClick}>
						<ContentAdd />
					</FloatingActionButton>
				</div>
			</div>
		);
		if(this.props.isLoading){
			return(
				<div>
					<p>Loading...</p>
				</div>
			);
		}
		/*
		If the Redux error state has changed(Default: false)
		Then Api returned response with an error 
		*/
		if(this.props.error !== false){
			console.log('You Got ',this.props.error);
			return (
				<Card key='1'>
					<CardHeader
						title="No Shoppinglist Items"
					/>
					<CardText>
						{this.props.error}
					</CardText>
					{fab}
				</Card>
			);
		}
		if(this.props.data.items.length === 0){
			return(
				<Card key='1'>
					<CardHeader
						title="No Items"
					/>
					<CardText>
						You currently Have No Items in Shoppinglist '{this.props.data.name}'
					</CardText>
					{fab}
				</Card>
			);
		}
		else{
			if(this.props.isOpenUpdateItem){
				return(
					<UpdateShoppingListItem {...this.props}/>
				);
			}
			this.props.data.items.map((item)=>
				cards.push(
					<Card key={item.id}
						style={{
							margin: '1% auto',
						}}
					>
						<CardHeader
							title={item.name}
							style={{
								textAlign: 'center',
								'fontStyle': 'italic',
								'fontWeight': 'bold',
							}}
						/>
						<CardText>
							Price KSH: {item.price} /=
						</CardText>
						<CardText>
							Quantity: {item.quantity}
						</CardText>
						<CardActions>
							<FlatButton label="Update" primary={true} onClick={()=>{this.props.activateUpdateItem(item);}}/>
							<FlatButton name='delete' label="Delete" secondary={true} onClick={()=>{this.confirmDelete(item.id);}}/>
						</CardActions>
					</Card>
				));
			/**
			 * Shoppinglist Delete confirmation
			 */
			let actions = [
				<FlatButton
					label="Cancel"
					primary={true}
					onClick={this.handleClose}
				/>,
				<FlatButton
					label="OK"
					primary={true}
					onClick={this.handleDelete}
				/>,
			];
			return(
				<div>
					<div>
					    Shoppinglist '{this.props.data.name}'
					</div>
					{fab}
					<Snackbar
						open={this.props.openSb}
						message={this.props.message}
						autoHideDuration={3000}
						onRequestClose={this.props.hideSnackBar}
					/>
					<Dialog
						actions={actions}
						modal={false}
						open={this.state.conf_delete}
						onRequestClose={this.handleClose}
					>
         			Confirm Delete?
					</Dialog>
				</div>
			);
		} 	
	}
}
function mapStateToProps(state){
	return {
		data: state.shoppinglist_items.data,
		isLoading: state.shoppinglist_items.isLoading,
		error: state.shoppinglist_items.error,
		openAddItem: state.shoppinglist_items.openAddItem,
		openSb: state.shoppinglist_items.openSb,
		isOpenUpdateItem: state.shoppinglist_items.isOpenUpdateItem,
		message: state.shoppinglist_items.message,
		item: state.shoppinglist_items.item,
	};
}
export default connect(mapStateToProps, {viewShoppingList, activateAddItem, activateUpdateItem, hideSnackBar, deleteShoppinglistItem, resetErrors})(ShoppingListItems);