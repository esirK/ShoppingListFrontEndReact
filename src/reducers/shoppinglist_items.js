import * as types from '../constants/actiontypes';

const initState = {
	data: {
		items: []
	},
	isLoading: false,
	isAdding: false,
	isDeleting: false,
	id:-1,
	message: '',
	error: false,
	openAddItem: false,
	openSb: false,	
	isOpenUpdateItem: false,	
	item:{},
};

export default (state= initState, action)=>{
	switch(action.type){
	case types.ACTIVATE_ADD_ITEM:
		//Add new Item FAB has been clicked
		return{
			...state, openAddItem: true
		};
	case types.DEACTIVATE_ADD_ITEM:
		//Add new Item FAB has been clicked
		return{
			...state, openAddItem: false
		};
	case types.ACTIVATE_UPDATE_ITEM:
		//Change state to show the Update Item dialog
		return{
			...state, isOpenUpdateItem: true, item: action.item
		};
	case types.DEACTIVATE_UPDATE_ITEM:
		//Change state to remove the Update Item dialog
		return{
			...state, isOpenUpdateItem: false, isUpdating: false
		};
	case types.ADDING_SHOPPINGLIST_ITEM_STARTED:
		return{
			...state, isAdding: true
		};
	case types.SHOPPINGLIST_ITEM_ADDED:
		return{
			...state, isAdding: false, openAddItem: false, openSb:true,message: action.message,
		};
	case types.DELETING_ITEM_STARTED:
		return{
			...state, isDeleting: true
		};
	case types.ITEM_DELETED_SUCCESSFULLY:
		return{
			...state, isDeleting: false, message: action.message, openSb:true,
		};
	case types.SHOPPINGLIST_ITEMS_LOADED_SUCCESSFULLY:
		return{
			...state, data: action.data, id:action.id, isLoading: false
		};
	case types.SHOPPINGLIST_ITEMS_LOADING:
		return{
			...state, isLoading: true
		};
	case types.SHOPPINGLIST_ITEMS_FAILED:
		return{
			...state, isLoading: false, error: action.error
		};
	case types.ERROR_ENCOUNTERED:
		return{
			...state, error: action.error
		};
	case types.CLEAR_ERRORS:
		return{
			...state, error: false, openSb: false
		};
	case types.HIDE_SNACK_BAR:
		return{
			...state,openSb: false
		};
	default:
		return state;
	}
};