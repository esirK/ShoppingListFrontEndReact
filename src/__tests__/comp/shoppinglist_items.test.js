import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {ShoppingListItems} from '../../components/shoppinglist_items';

describe('Render Shopping List items',()=>{
	const props = {
		activateAddItem:jest.fn(),
		activateUpdateItem:jest.fn(),
		deleteShoppinglistItem:jest.fn(),
		hideSnackBar:jest.fn(),
		error:false,
		addFab:false,
		isOpenUpdateItem:false,
		match:{
			isExact: true,
			params:{id:166},
			path:'/:id/shoppinglist_items',
			url:'/166/shoppinglist_items',
		},
		data:{
			items:[]
		},
		message:'',
		openAddItem:'',
		openSb:false,
		isLoading: false,
		resetErrors: jest.fn(),
		viewShoppingList: jest.fn(),
	};
    
	const shoppinglistItems = shallow(<ShoppingListItems {...props}/>);
	it('Renders ShoppingListItems component', ()=>{
		expect(shallowToJson(shoppinglistItems)).toMatchSnapshot();
	});
});