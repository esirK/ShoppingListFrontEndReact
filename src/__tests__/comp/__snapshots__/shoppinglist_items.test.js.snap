// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Shopping List items Renders ShoppingListItems component 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
  key="1"
>
  <CardHeader
    avatar={null}
    title="No Items"
  />
  <CardText>
    You currently Have No Items in Shoppinglist '
    '
  </CardText>
  <div
    id="cards"
  >
    <div
      id="fab"
    >
      <FloatingActionButton
        disabled={false}
        mini={false}
        onClick={[Function]}
        secondary={true}
        zDepth={2}
      >
        <ContentAdd />
      </FloatingActionButton>
    </div>
  </div>
</Card>
`;
